registers/src/csrng.rs

```diff
@@ -4,16 +4,48 @@
 //
 #![allow(clippy::erasing_op)]
 #![allow(clippy::identity_op)]
+/// A zero-sized type that represents ownership of this
+/// peripheral, used to get access to a Register lock. Most
+/// programs create one of these in unsafe code near the top of
+/// main(), and pass it to the driver responsible for managing
+/// all access to the hardware.
+pub struct CsrngReg {
+    _priv: (),
+}
+impl CsrngReg {
+    pub const PTR: *mut u32 = 0x20002000 as *mut u32;
+    /// # Safety
+    ///
+    /// Caller must ensure that all concurrent use of this
+    /// peripheral in the firmware is done so in a compatible
+    /// way. The simplest way to enforce this is to only call
+    /// this function once.
+    pub unsafe fn new() -> Self {
+        Self { _priv: () }
+    }
+    /// Returns a register block that can be used to read
+    /// registers from this peripheral, but cannot write.
+    pub fn regs(&self) -> RegisterBlock<ureg::RealMmio> {
+        RegisterBlock {
+            ptr: Self::PTR,
+            mmio: core::default::Default::default(),
+        }
+    }
+    /// Return a register block that can be used to read and
+    /// write this peripheral's registers.
+    pub fn regs_mut(&mut self) -> RegisterBlock<ureg::RealMmioMut> {
+        RegisterBlock {
+            ptr: Self::PTR,
+            mmio: core::default::Default::default(),
+        }
+    }
+}
 #[derive(Clone, Copy)]
-pub struct RegisterBlock<TMmio: ureg::Mmio + core::borrow::Borrow<TMmio> = ureg::RealMmio> {
+pub struct RegisterBlock<TMmio: ureg::Mmio + core::borrow::Borrow<TMmio>> {
     ptr: *mut u32,
     mmio: TMmio,
 }
-impl RegisterBlock<ureg::RealMmio> {
-    pub fn csrng_reg() -> Self {
-        unsafe { Self::new(0x20002000 as *mut u32) }
-    }
-}
+impl RegisterBlock<ureg::RealMmioMut<'_>> {}
```

Comment:
We don't need this empty `impl` anymore.

Reply:
Fixed in [58353e6](https://github.com/chipsalliance/caliptra-sw/pull/277/commits/58353e6bb190ea2e4ba9421525f3fa88463333c3)